### PR TITLE
docs: restructure and update "Deno 1.x to 2.x Migration Guide"

### PR DESCRIPTION
**Pull request by iuioiua**
Updated to align with ongoing work. See https://github.com/denoland/deno/issues/22079 for reference.

### CODE REVIEW

File: runtime/manual/advanced/migrate_deprecations.md [hunk 1]
```diff
@@ -12,11 +12,96 @@ will continue to be updated until the launch of 2.0.
...
+
+### deno cache
+
+The `deno cache` command has been merged into the `deno install` command under
+the `--entrypoint` option.
+
+```diff
+- deno cache main.ts
++ deno install --entrypoint main.ts
+```
```
Comment (iuioiua) [inline runtime/manual/advanced/migrate_deprecations.md:46]: Actually, `deno cache` is being deprecated in Deno 2, not removed. I'll remove this section.

Comment (bartlomieju) [reply runtime/manual/advanced/migrate_deprecations.md:46]: We should not use `deno cache` anymore in any docs - it's getting deprecated, but we don't want to advertise it in any way.

Comment (iuioiua) [reply runtime/manual/advanced/migrate_deprecations.md:46]: Removed

Comment (marvinhagemeister) [reply runtime/manual/advanced/migrate_deprecations.md]: @iuioiua Why did you remove it? We _do_ want to move people over to `deno install --entrypoint` from `deno cache`

File: runtime/manual/advanced/migrate_deprecations.md [hunk 2]
```diff
@@ -2,8 +2,8 @@
 title: "Deno 1.x to 2.x Migration Guide"
 ---
 
-This document contains guidance for migrating from Deno version 1.x to the
-upcoming Deno 2.0.
+This document contains guidance for migrating from Deno 1.x to 2.x, namely for
+subcommands, flags and symbols that have been removed or changed in Deno 2.
 
 :::caution Work in progress
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:8]: Can this be removed now?

File: runtime/manual/advanced/migrate_deprecations.md [hunk 3]
```diff
@@ -12,11 +12,86 @@ will continue to be updated until the launch of 2.0.
 
 :::
 
-## API changes and deprecations
+## Subcommands
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:15]:
```suggestion
## Updated subcommands
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 4]
```diff
@@ -12,11 +12,86 @@ will continue to be updated until the launch of 2.0.
...
+The `deno vendor` command has been replaced by a `"vendor": true` configuration
+option in `deno.json`.
+
+```json title="deno.json"
+{
+  "vendor": true
+}
+```
+
+## Flags
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:49]:
```suggestion
## Updated flags
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 5]
```diff
@@ -12,11 +12,86 @@ will continue to be updated until the launch of 2.0.
...
+### `--trace-ops`
+
+Use the `--trace-leaks` CLI flag instead.
+
+```diff
+- deno test --trace-ops
++ deno test --trace-leaks
+```
+
+## Symbols
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:94]:
```suggestion
## Updated symbols
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 6]
```diff
@@ -70,28 +155,26 @@ Use [`Deno.Conn`](https://docs.deno.com/api/deno/~/Deno.Conn) instance methods
...
 + await conn.closeWrite();
+
+- Deno.close(conn.rid);
++ conn.close();
 ```
 
 See the [Deno 1.40 blog post][Deno 1.40 blog post] for details.
 
-### `Deno.ListenTlsOptions.certChain`
+### `Deno.ConnectTlsOptions.certChain`
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:177]:
```suggestion
### Deno.ConnectTlsOptions.certChain
```

avoid using backticks in titles, it renders and reads poorly

File: runtime/manual/advanced/migrate_deprecations.md [hunk 7]
```diff
@@ -12,11 +12,86 @@ will continue to be updated until the launch of 2.0.
...
+Use the `--trace-leaks` CLI flag instead.
+
+```diff
+- deno test --trace-ops
++ deno test --trace-leaks
+```
+
+## Symbols
 
 ### `Deno.Buffer`
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:96]:
```suggestion
### Deno.Buffer
```

no backticks in titles

File: runtime/manual/advanced/migrate_deprecations.md [hunk 8]
```diff
@@ -559,6 +718,25 @@ instead.
   console.log(new TextDecoder().decode(stdout));
 ```
 
+Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
+its implementation remains, in order to reduce breaking changes. So
+alternatively, you can ignore "property does not exist" TypeScript errors by
+using the
+[`@ts-ignore`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments)
+directive.
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:726]:
```suggestion
:::note
This symbol is soft-removed as of Deno 2. Its types have been removed but
its implementation remains, in order to reduce breaking changes. So
alternatively, you can ignore "property does not exist" TypeScript errors by
using the
[`@ts-ignore`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments)
directive.
:::
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 9]
```diff
@@ -559,6 +718,25 @@ instead.
...
+
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:727]:
```suggestion
Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
its implementation remains in order to reduce breaking changes. 
You can ignore "property does not exist" TypeScript errors by
using the
[`@ts-ignore`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments)
directive.

```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 10]
```diff
@@ -601,6 +783,23 @@ Use [`Deno.serve()`](https://docs.deno.com/api/deno/~/Deno.serve) instead.
 + Deno.serve({ port: 80 }, () => new Response("Hello World"));
 ```
 
+Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
+its implementation remains, in order to reduce breaking changes. So
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:787]:
```suggestion
its implementation remains in order to reduce breaking changes. So
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 11]
```diff
@@ -559,6 +718,25 @@ instead.
   console.log(new TextDecoder().decode(stdout));
 ```
 
+Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
+its implementation remains, in order to reduce breaking changes. So
+alternatively, you can ignore "property does not exist" TypeScript errors by
+using the
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:724]:
```suggestion
Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
its implementation remains in order to reduce breaking changes. 
You can ignore "property does not exist" TypeScript errors by
using the
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 12]
```diff
@@ -638,18 +839,29 @@ Use [`Deno.stderr`](https://docs.deno.com/api/deno/~/Deno.stderr) instance
...
 
-```diff
 - Deno.close(Deno.stderr.rid);
 + Deno.stderr.close();
 ```
 
+Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
+its implementation remains, in order to reduce breaking changes. So
+alternatively, you can ignore "property does not exist" TypeScript errors by
+using the
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:858]:
```suggestion
Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
its implementation remains in order to reduce breaking changes. 
You can ignore "property does not exist" TypeScript errors by
using the
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 13]
```diff
@@ -682,18 +905,29 @@ Use [`Deno.stdout`](https://docs.deno.com/api/deno/~/Deno.stdout) instance
...
 
-```diff
 - Deno.close(Deno.stdout.rid);
 + Deno.stdout.close();
 ```
 
+Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
+its implementation remains, in order to reduce breaking changes. So
+alternatively, you can ignore "property does not exist" TypeScript errors by
+using the
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:924]:
```suggestion
Note: This symbol is soft-removed as of Deno 2. Its types have been removed but
its implementation remains in order to reduce breaking changes. 
You can ignore "property does not exist" TypeScript errors by
using the
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 14]
```diff
@@ -882,8 +1123,21 @@ Use [`Deno.openSync()`](https://docs.deno.com/api/deno/~/Deno.openSync) or
 
 ```diff
 - const file = new Deno.FsFile(3);
-+ const file = await Deno.open("/foo/bar.txt");
++ using file = await Deno.open("/foo/bar.txt");
 ```
 
+### `window`
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:1129]:
```suggestion
### window
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 15]
```diff
@@ -12,11 +12,86 @@ will continue to be updated until the launch of 2.0.
...
+
+```json title="deno.json"
+{
+  "vendor": true
+}
+```
+
+## Flags
+
+### `--allow-none`
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:51]:
```suggestion
### --allow-none
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 16]
```diff
@@ -12,11 +12,86 @@ will continue to be updated until the launch of 2.0.
...
+
+Use the `--permit-no-files` CLI flag instead.
+
+```diff
+- deno test --allow-none
++ deno test --permit-no-files
+```
+
+### `--jobs`
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:60]:
```suggestion
### --jobs
```

File: runtime/manual/advanced/migrate_deprecations.md [hunk 17]
```diff
@@ -12,11 +12,86 @@ will continue to be updated until the launch of 2.0.
...
+Use the
+[`DENO_JOBS`](https://docs.deno.com/runtime/manual/basics/env_variables/#special-environment-variables)
+environment variable instead.
+
+```diff
+- deno test --jobs=4 --parallel
++ DENO_JOBS=4 deno test --parallel
+```
+
+### `--ts`
```
Comment (thisisjofrank) [inline runtime/manual/advanced/migrate_deprecations.md:71]:
```suggestion
### --ts
```

File: runtime/manual/advanced/migrate_deprecations.md (hunk 1 above)
Comment (iuioiua) [reply runtime/manual/advanced/migrate_deprecations.md:46]: Sorry. Misunderstood. Added back now.

File: runtime/manual/advanced/migrate_deprecations.md (hunk 15 above)
Comment (iuioiua) [reply runtime/manual/advanced/migrate_deprecations.md:61]: AFAIK, symbols and parts of CLI commands are always expressed as inline code.

Comment (iuioiua) [reply runtime/manual/advanced/migrate_deprecations.md:61]: Sorry, just saw your comment below. Ignore.

**Review by thisisjofrank — APPROVED** [review]

**Pull request merged**